lib/warden/proxy.rb

```diff
@@ -29,6 +29,11 @@ def initialize(env, manager) #:nodoc:
       @env, @users, @winning_strategies, @locked = env, {}, {}, false
       @manager, @config = manager, manager.config.dup
       @strategies = Hash.new { |h,k| h[k] = {} }
+    end
+
+    # Run the on_request callbacks
+    # :api: public
+    def on_request
```

Comment:
I think this can be private, but I'll leave that up to the maintainers.

Reply:
At first I want to make it private, so that I could be lazy without writing tests, docs, etc. However if we're making this private, we probably need to write like `env['warden'].__send__(:on_request)`.

I am ok with this, but I am not sure if it's fine. (We don't have friends in Ruby 😓)

Reply:
I meant `:api: private` in this case, and then naming the method `#_on_request`

I think that's how the methods that aren't part of the public api of warden are indicated. But I'm not sure that's the actual convention in this project 😄.

Reply:
Oh! You're right, I didn't think of that it could be `:api: private` yet it's a public method in Ruby. I'll make the change, but maybe still call it `on_request` because the other parts weren't using `_` as the prefix, and those with them have a counterpart method.